### PR TITLE
trivial: finish all components from a cab and show errors

### PR DESCRIPTION
**Pull request by CragW**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-release.c [hunk 1]
```diff
@@ -940,6 +940,9 @@ fu_release_load(FuRelease *self,
 					   self);
 	}
 
+	/* don't exercise requirement checks for get-details */
+	return TRUE;
+
```
Comment (superm1) [inline src/fu-release.c:945]: don't you need some sort of check here?

**Comment by CragW** [comment]
How to understand it's called by `get-details` and don't have to run requirement checks? Or, a better way to fix #4783?

File: src/fu-release.c (hunk 1 above)
Comment (CragW) [reply src/fu-release.c:945]: Yes it should, though the load-releases is share code between install tasks and get-details / get-updates it seems harder to make a check here. Ideas is appreciated.

**Comment by hughsie** [comment]
Can you add a self test that fails? i.e. with the exact but least complicated scenario?

File: src/fu-release.c (hunk 1 above)
Comment (hughsie) [reply src/fu-release.c:945]: I think the idea was we'd do something like:

	request = fu_engine_request_new(FU_ENGINE_REQUEST_KIND_ONLY_SUPPORTED);
	fu_engine_request_set_feature_flags(request, ~0);

but that's not happening for get-details. Will ponder.

File: src/fu-release.c [hunk 2]
```diff
@@ -942,6 +942,7 @@ fu_release_load(FuRelease *self,
 
 	/* check requirements for device */
 	if (self->device != NULL && self->request != NULL &&
+	    (install_flags & FWUPD_INSTALL_FLAG_NONE) != 0 &&
```
Comment (superm1) [inline src/fu-release.c:945]: I think I would feel better about this to have a new install flag for ignore requirements explicitly set by the get_details flow.  There may be multiple callers into this function that might want to evaluate requirements even if INSTALL_FLAG_NONE wasn't set.

Comment (CragW) [reply src/fu-release.c]: I've added `FWUPD_INSTALL_FLAG_NO_REQUIREMENTS` install flag and introduced it from 1.8.2 onward.

**Pull request closed without merging**